test/index.js

```diff
@@ -1793,6 +1793,42 @@ describe('normalizr', function () {
     });
   });
 
+  it('can normalize iterables keyed with their id', function () {
+    var user = new Schema('users', {
+      idAttribute: function(obj, key) {
+        return key;
+      }
+    })
+
+    var input = {
+      1: {
+        name: 'Adam'
+      },
+      4: {
+        name: 'Jeremy'
+      }
+    };
+
+    Object.freeze(input);
+
+    normalize(input, valuesOf(user)).should.eql({
+      result: {
+        1: '1',
+        4: '4'
+      },
```

Comment:
I'd like to see the result be `[1, 4]` as well, since an array is the typical result format. What work is required for that?